doped/utils/parsing.py

```
@@ -1988,3 +2009,374 @@ def _multiple_files_warning(file_type, directory, chosen_filepath, action=None,
+        comp_dict = composition.as_data_dict()['unit_cell_composition']
+        
+        elements = [k.name for k in struct.elements] 
+        n_i = np.array(list(comp_dict.values()))
+        u_i = np.array([cls._get_element_formation_energy(elem) for elem in elements])
+        
+        std_form_energy = (e_bulk - np.sum(n_i*u_i))/np.sum(n_i)
+
+        return std_form_energy
+    
+    @classmethod
+    def _get_element_formation_energy(cls, 
```

Ruff (0.12.2)

Trailing whitespace

Remove trailing whitespace

(W291)